bin/estrip

```diff
@@ -0,0 +1,458 @@
+		done
+		unset lockfile
+	fi
+
+	[ -f "${inode_link}_stripped" ] && already_stripped=true || already_stripped=false
+
+	if ! ${already_stripped} ; then
+		if ${PRESERVE_XATTR} ; then
+			xt_data=$(dump_xattrs "${x}")
+		fi
+		save_elf_sources "${x}"
+	fi
+
+	if ${strip_this} ; then
+
+		# see if we can split & strip at the same time
+		if [[ -n ${SPLIT_STRIP_FLAGS} ]] ; then
+			local shortname="${x##*/}.debug"
+			local splitdebug="${tmpdir}/splitdebug/${shortname}.${BASHPID:-$(__bashpid)}"
+			${already_stripped} || \
+			${STRIP} ${strip_flags} \
+				-f "${splitdebug}" \
+				-F "${shortname}" \
+				"${x}"
+			save_elf_debug "${x}" "${inode_link}_debug" "${splitdebug}"
+		else
+			save_elf_debug "${x}" "${inode_link}_debug"
+			${already_stripped} || \
+			${STRIP} ${strip_flags} "${x}"
+		fi
+	fi
+
+	if ${already_stripped} ; then
+		rm -f "${x}" || die "rm failed unexpectedly"
+		ln "${inode_link}_stripped" "${x}" || die "ln failed unexpectedly"
+	else
+		ln "${x}" "${inode_link}_stripped" || die "ln failed unexpectedly"
+		if [[ ${xt_data} ]] ; then
+			restore_xattrs <<< "${xt_data}"
+		fi
+	fi
+
+	[[ -n ${lockfile} ]] && rm -f "${lockfile}"
+}
+
+# The existance of the section .symtab tells us that a binary is stripped.
+# We want to log already stripped binaries, as this may be a QA violation.
+# They prevent us from getting the splitdebug data.
+if ! ${RESTRICT_binchecks} && ! ${RESTRICT_strip} ; then
+	# We need to do the non-stripped scan serially first before we turn around
+	# and start stripping the files ourselves.  The log parsing can be done in
+	# parallel though.
+	log=${tmpdir}/scanelf-already-stripped.log
+	scanelf -yqRBF '#k%F' -k '!.symtab' "$@" | sed -e "s#^${ED%/}/##" > "${log}"
+	(
+	__multijob_child_init
+	qa_var="QA_PRESTRIPPED_${ARCH/-/_}"
+	[[ -n ${!qa_var} ]] && QA_PRESTRIPPED="${!qa_var}"
+	if [[ -n ${QA_PRESTRIPPED} && -s ${log} && \
+		${QA_STRICT_PRESTRIPPED-unset} = unset ]] ; then
+		shopts=$-
+		set -o noglob
+		for x in ${QA_PRESTRIPPED} ; do
+			sed -e "s#^${x#/}\$##" -i "${log}"
+		done
+		set +o noglob
+		set -${shopts}
+	fi
+	sed -e "/^\$/d" -e "s#^#/#" -i "${log}"
+	if [[ -s ${log} ]] ; then
+		__vecho -e "\n"
+		eqawarn "QA Notice: Pre-stripped files found:"
+		eqawarn "$(<"${log}")"
+	else
+		rm -f "${log}"
+	fi
+	) &
+	__multijob_post_fork
+fi
+
+# Since strip creates a new inode, we need to know the initial set of
+# inodes in advance, so that we can avoid interference due to trying
+# to strip the same (hardlinked) file multiple times in parallel.
+# See bug #421099.
+if  [[ ${USERLAND} == BSD ]] ; then
+	get_inode_number() { stat -f '%i' "$1"; }
+else
+	get_inode_number() { stat -c '%i' "$1"; }
+fi
+cd "${tmpdir}/inodes" || die "cd failed unexpectedly"
+while read -r x ; do
+	inode_link=$(get_inode_number "${x}") || die "stat failed unexpectedly"
+	echo "${x}" >> "${inode_link}" || die "echo failed unexpectedly"
+done < <(
+	# Use sort -u to eliminate duplicates for bug #445336.
+	(
```

Comment:
But now you're not using the results of `--queue` and instead stripping all files again…

Reply:
Thanks, updated to use this code conditionally:
```bash
while IFS= read -d '' -r x ; do
	rm -f "${x}" || die
	inode_link=$(get_inode_number "${x%.estrip}") || die "stat failed unexpectedly"
	echo "${x%.estrip}" >> "${inode_link}" || die "echo failed unexpectedly"
done < <(find "${ED}" -name '*.estrip' -print0)
```